python/ray/data/_internal/compute.py

```diff
@@ -27,14 +28,23 @@
 T = TypeVar("T")
 U = TypeVar("U")
 
+
+@dataclass
+class TaskContext:
```

Comment:
Shall we move this to `execution/interfaces.py` (in anticipation of us deleting compute.py?)

Reply:
+1 for move. We plan to eventually delete this file.